src/clj/rems/api/services/resource.clj

```diff
@@ -28,25 +31,12 @@
 
 (defn set-resource-archived! [{:keys [id archived]}]
   (util/check-allowed-organization! (:organization (get-resource id)))
-  (let [catalogue-items
-        (->> (catalogue/get-localized-catalogue-items {:resource-id id
-                                                       :archived false})
-             (map #(select-keys % [:id :title :localizations])))
-        licenses (licenses/get-resource-licenses id)
-        archived-licenses (filter :archived licenses)]
-    (cond
-      (and archived (seq catalogue-items))
-      {:success false
-       :errors [{:type :t.administration.errors/resource-in-use
-                 :catalogue-items catalogue-items}]}
-
-      (and (not archived) (seq archived-licenses))
-      {:success false
-       :errors [{:type :t.administration.errors/license-archived
-                 :licenses archived-licenses}]}
-
-      :else
-      (do
-        (db/set-resource-archived! {:id id
-                                    :archived archived})
-        {:success true}))))
+  (if-let [errors (if archived
+                    (dependencies/archive-errors :t.administration.errors/resource-in-use {:resource/id id})
+                    (dependencies/unarchive-errors {:resource/id id}))]
+    {:success false
+     :errors errors}
+    (do
+      (db/set-resource-archived! {:id id
+                                  :archived archived})
+      {:success true})))
```

Comment:
These lines of code are not quite consistent between the different items but I wonder if that is necessary. Seems like there is some boilerplate here that could be removed.

Reply:
Yeah now the code depends on whether unarchive-errors are possible. I'll refactor this in the next PR when I simplify the error structures.